### PR TITLE
Weekly dependency updates for Gemfile.lock

### PR DESCRIPTION
**Pull request by jmartin-tech**
This PR should have been auto-generated by [actions/github-script](https://github.com/actions/github-script). `bundle update` revealed the following gems have new version to be evaluated for update.

More adjustment to get automation are in progress, for now let us pickup here this week.

## Verification

List the steps needed to make sure this thing works

- [x] Security review diffs for updated gems

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -185,29 +185,10 @@ GEM
...
-      faraday-em_synchrony (~> 1.0)
-      faraday-excon (~> 1.1)
-      faraday-httpclient (~> 1.0)
-      faraday-multipart (~> 1.0)
-      faraday-net_http (~> 1.0)
-      faraday-net_http_persistent (~> 1.0)
-      faraday-patron (~> 1.0)
-      faraday-rack (~> 1.0)
-      faraday-retry (~> 1.0)
+    faraday (2.3.0)
```
Comment (jmartin-r7) [inline Gemfile.lock:188]: While the version bump here is significant it looks to be reasonable and is based on update of dependencies for `octokit`.

**Review by gwillcox-r7 — APPROVED** [review]

**Comment by gwillcox-r7** [comment]
Going to need to rebase due to me updating a few things on a previous PR but will get that sorted out and then land this.

File: Gemfile.lock (hunk 1 above)
Comment (adfoster-r7) [reply Gemfile.lock:188]: Looks like this update now outputs to msfconsole on bootup by default which is annoying 😄 
```
$ bundle exec ruby ./msfconsole -q
To use retry middleware with Faraday v2.0+, install `faraday-retry` gem <-------
[*] Using configured payload windows/meterpreter/reverse_tcp
msf6 exploit(windows/smb/kerberos_test) > 
```

Tracked it down to octokit:
https://github.com/octokit/octokit.rb/blame/dd622a38073f383e7e355f6897ffed60b6198926/lib/octokit/default.rb#L7-L15

We can silence this output with OCTOKIT_SILENT:
https://github.com/octokit/octokit.rb/blob/94925391160987c74bd0e4dcc221255e1f0ba842/lib/octokit/warnable.rb#L8-L16

But that seems like a workaround, I'm tempted to say Octokit is either logging out for behavior we don't explicitly rely on - or if there's now a subtle bug for the code that relies on Octokit's on retry functionality. Would have to poke further. I believe it's only the pull request finder that uses this functionality.

Comment (adfoster-r7) [reply Gemfile.lock:188]: Yeah, looks like we'd be dropping the default retry behavior that used to be present - and now it explicitly requires a separate gem dependency to work:

https://github.com/octokit/octokit.rb/blame/ff8dc9c181e84de008c973db1745a65d895c1f9c/lib/octokit/default.rb#L33-L37

Comment (adfoster-r7) [reply Gemfile.lock:188]: PR to add the gem: https://github.com/rapid7/metasploit-framework/pull/16662

**Pull request merged**